### PR TITLE
moved locale patching methods to LanguageUtils class

### LINKED ISSUE #4235: Check Locale.getDefault() usage
Check the conversation [here for context](https://github.com/wordpress-mobile/WordPress-Android/pull/4234#issuecomment-226190098)

1) We need to check all the places where `Locale.getDefault()` is used and make sure we use `RestClientUtils.patchDeviceLanguageCode()` instead, wherever needed.

2) Also, for a more grained control I'd suggest refactoring  `RestClientUtils.getRestLocaleParams()` to make use of the _currently configured locale_, provided by `context.getResources().getConfiguration().locale.toString()`  rather than `Locale.getDefault()` which is only updated upon application launch. Suggestion: use a new class `LanguageUtils` with methods `getDeviceLanguage` and `patchDeviceLanguage`

### PR DESCRIPTION
**Pull request by mzorz**
Fixes #4235 

To test: There are several places where the code is used now, so I would suggest trying two sets of functionalities:
A) Changes made in this changset: Change your device language settings to a "patched" language such as hebrew. Then create a new user/blog. If it all works correctly you should be able to go to the blog's settings and verify it's set to hebrew

B) Changes already made before but moved to the new LanguageUtils class: Try going to Stats and see if the content is served in the language of choice.

Needs review: @maxme

### CODE REVIEW

**Comment by maxme** [comment]
We have a few source library dependencies in the app: `Analytics`, `Editor`, `Networking` and `Utils`. One of these dependencies is to be reused in other apps. `Utils` is a compilation of static methods to help with various Android or Java things.

When we write an utility class that is not WordPress dependent, we should put it in `Utils`. That's the case for `LanguageUtils`, it should go to `WordPress-Android/libs/utils/WordPressUtils/src/main/java/org/wordpress/android/util/`.

File: libs/networking/WordPressNetworking/src/main/java/org/wordpress/android/networking/LanguageUtils.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+import java.util.HashMap;
+import java.util.Locale;
+
+/**
+ * Methods for dealing with i18n messages
+ */
+public class LanguageUtils {
+
+    public static String getCurrentDeviceLanguage(Context context) {
```
Comment (maxme) [inline libs/networking/WordPressNetworking/src/main/java/org/wordpress/android/networking/LanguageUtils.java]: We should rename this function `getCurrentDeviceLanguageCode` and write a `getCurrentDeviceLocale` that returns a `Locale` object. Then we can reuse it in other places like `GeocoderUtils` for instance.

Comment (mzorz) [inline libs/networking/WordPressNetworking/src/main/java/org/wordpress/android/networking/LanguageUtils.java]: Did this and changed references properly wherever `Locale.getDefault()` was used (_and_ made sense to use the new method)

**Comment by maxme** [comment]
:shipit:

**Pull request merged**